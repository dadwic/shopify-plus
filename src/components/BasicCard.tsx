import * as React from "react";
import { styled } from "@mui/system";
import Box from "@mui/material/Box";
import Card from "@mui/material/Card";
import CardMedia from "@mui/material/CardMedia";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";

interface Props {
  title: string;
  price: string;
  src: string;
}

const Circle = styled("span")`
  height: 12px;
  width: 12px;
  cursor: auto;
  background-color: ${(props) => props.color};
  border-radius: 50%;
  display: inline-block;
  margin-left: 8px;
  @media (max-width: 599px) {
    margin-left: 1px;
    margin-right: 7px;
  }
  &.active {
    border: 1.5px solid #fff;
    box-shadow: 0 0 0 1px #000;
  }
`;

export default function BasicCard({ title, price, src }: Props) {
  return (
    <Card
      elevation={0}
      sx={{
        mr: { xs: 1.5, sm: 4 },
        flex: "0 0 auto",
        width: { xs: 140, sm: 230 },
      }}
    >
      <CardMedia component="img" image={src} alt={title} />
      <CardContent sx={{ p: 0 }}>
        <Typography
          fontFamily="Avenir"
          sx={{ fontSize: "12.8px", mt: 1.5, mb: 0.5 }}
        >
          {title}
        </Typography>
        <Box
          sx={{
            display: "flex",
            flexDirection: { xs: "column", sm: "row" },
            justifyContent: "space-between",
          }}
        >
          <Typography
            sx={{ fontSize: "12.8px" }}
            fontWeight={500}
            fontFamily="Avenir"
          >
            {price}
          </Typography>
          <div>
            <Circle color="#99C3CC" className="active" />
            <Circle color="#CC9999" />
            <Circle color="#CB99CC" />
            <Circle color="#A6CC99" />
          </div>
        </Box>
      </CardContent>
    </Card>
  );
}
