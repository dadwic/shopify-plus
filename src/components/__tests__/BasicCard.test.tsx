import React from "react";
import renderer from "react-test-renderer";
import { productsMock } from "__mocks__";
import BasicCard from "../BasicCard";

describe("BasicCard", () => {
  it("should correctly match snapshot", () => {
    const tree = renderer.create(<BasicCard {...productsMock[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
