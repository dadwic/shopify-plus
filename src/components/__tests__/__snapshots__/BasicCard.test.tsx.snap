// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicCard should correctly match snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-f7gmfx-MuiPaper-root-MuiCard-root"
>
  <img
    alt="365 Signature Hoodie"
    className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
    src="/img/bg-1.jpg"
  />
  <div
    className="MuiCardContent-root css-1i9wt8r-MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-jqu9gp-MuiTypography-root"
    >
      365 Signature Hoodie
    </p>
    <div
      className="MuiBox-root css-asaamw"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1j97s3x-MuiTypography-root"
      >
        â‚¬33.95
      </p>
      <div>
        <span
          className="active css-111dolv"
          color="#99C3CC"
        />
        <span
          className="css-16lu8s2"
          color="#CC9999"
        />
        <span
          className="css-1tubl60"
          color="#CB99CC"
        />
        <span
          className="css-988mqq"
          color="#A6CC99"
        />
      </div>
    </div>
  </div>
</div>
`;
